### PR TITLE
Fix removal of offline runners.

### PR DESCRIPTION
**Pull request by yhaliaw**
Fixing a bug causing offline runners to be not removed from GitHub.

### CODE REVIEW

File: src/runner_manager.py [hunk 1]
```diff
@@ -422,3 +437,43 @@ def create_runner_info(
...
+        if isinstance(self.config.path, GitHubRepo):
+            token = self._clients.github.actions.create_registration_token_for_repo(
+                owner=self.config.path.owner, repo=self.config.path.repo
+            )
+        elif isinstance(self.config.path, GitHubOrg):
+            token = self._clients.github.actions.create_registration_token_for_org(
+                org=self.config.path.org
+            )
+        else:
+            assert_never(token)
```
Comment (jdkandersson) [inline src/runner_manager.py:457]: What does this do out of curiosity?

File: tests/unit/mock.py [hunk 2]
```diff
@@ -156,10 +159,24 @@ def list_runner_applications_for_org(self, org: str):
         return self._list_runner_applications()
 
     def create_registration_token_for_repo(self, owner: str, repo: str):
-        return RegisterToken({"token": "test registration token"})
+        return RegistrationToken(
+            {"token": "test registration token", "expires_at": "2020-01-22T12:13:35.123-08:00"}
```
Comment (jdkandersson) [inline tests/unit/mock.py:163]: Perhaps use secrets.hex_token?

File: tests/unit/test_runner.py [hunk 3]
```diff
@@ -131,7 +129,7 @@ def test_remove(
     """
 
     runner.create("test_image", vm_resources, binary_path, "test_token")
```
Comment (jdkandersson) [inline tests/unit/test_runner.py:131]: perhaps use secrets.hex_token for test tokens? Also applies for other tokens

File: src/runner_manager.py (hunk 1 above)
Comment (yhaliaw) [reply src/runner_manager.py:457]: It assert to the type system that this branch in the if-else statement can never be reached.

This is a new feature in Python 3.11 or typing-extensions 4.1:
https://docs.python.org/3/library/typing.html#typing.assert_never

The link includes an example usage:
https://typing.readthedocs.io/en/latest/source/unreachable.html

File: tests/unit/mock.py [hunk 4]
```diff
@@ -156,10 +159,24 @@ def list_runner_applications_for_org(self, org: str):
...
+
+    def create_remove_token_for_repo(self, owner: str, repo: str):
+        return RemoveToken(
+            {"token": "test registration token", "expires_at": "2020-01-22T12:13:35.123-08:00"}
+        )
+
+    def create_remove_token_for_org(self, org: str):
+        return RemoveToken(
+            {"token": "test registration token", "expires_at": "2020-01-22T12:13:35.123-08:00"}
+        )
```
Comment (yanksyoon) [inline tests/unit/mock.py]: Would it be simpler to have a mock remove token as a constant defined somewhere? :) It seems like it is getting used in a few places.

**Review by yanksyoon — APPROVED** [review]
Minor suggestions :) LGTM!

File: tests/unit/mock.py (hunk 4 above)
Comment (yhaliaw) [reply tests/unit/mock.py]: I updated to use a test fixture that generates the token with `secrets.token_hex`.

File: tests/unit/mock.py (hunk 2 above)
Comment (yhaliaw) [reply tests/unit/mock.py]: Updated

**Review by weiiwang01 — APPROVED** [review]

**Pull request merged**